apps/src/code-studio/pd/application/teacher1920/Teacher1920Application.jsx

```diff
@@ -93,4 +94,95 @@ export default class Teacher1920Application extends FormController {
     ga('set', 'page', url);
     ga('send', 'pageview');
   }
+
+  /**
+   * @override
+   */
+  setPage(i) {
+    const newPage = Math.min(
+      Math.max(i, 0),
+      this.getPageComponents().length - 1
+    );
+
+    this.setState({
+      currentPage: newPage
+    });
+
+    if (this.triedToSubmit) {
```

Comment:
Just learned: `FormController` has an `onSetPage` method designed to be overridden by a subclass.  It's called after mount and at the very end of the component update, if the page changed.  That might be a better place for this block to live, separating concerns a little bit.

https://github.com/code-dot-org/code-dot-org/blob/0d36cede970ed643522e30849b7690ea1b0f7888/apps/src/code-studio/pd/form_components/FormController.jsx#L198-L203

I think you still need to override `setPage` to skip the existing validation behavior, but the `if (this.triedToSubmit) {` and everything within it could be moved to the other handler.

Reply:
Sounds good. 1 question is `onSetPage` is called both on `componentWillMount` (even when page is reloaded) and `componentDidUpdate`. Is that what we want?

Reply:
Yes: I don't think it's a problem because the `this.triedToSubmit` check should keep us from rendering anything during `componentWillMount`.

Reply:
Fixed.